Replication manager's rep_start_pids now contains only {Tag, Pid} items

nickva opened the pull request:
Previously the local change feed was added to rep_start_pids as Pid only. So if
replication manager stopped and terminate/2 was called before that change
feed died, then

```
foreach(fun({_Tag, Pid}) -> ... end, [StartPids])
```

would crash with a function clause error.

Jira: COUCHDB-3082

nickva committed:
Replace hard-coded instances of <<"_replicator">> dbs with a macro

There are 3 of those. For now replace only those which refer to db names, not
role.

Jira: COUCHDB-3082

rnewson commented on src/couch_replicator_manager.erl:
```diff
@@ -182,7 +182,7 @@ init(_) ->
         max_retries = retries_value(
             config:get("replicator", "max_replication_retry_count", "10")),
-        rep_start_pids = [Pid],
+        rep_start_pids = [{rep_start, Pid}],
```
right idea but `Pid` here isn't a replication that's starting, it's the changes reader.

rnewson commented:
+1 but please make two commits. 1) DRY the replicator value, 2) fix the bug

nickva committed:
Replication manager's rep_start_pids now contains only {Tag, Pid} items

Previously the local change feed was added to rep_start_pids as Pid only. So if
replication manager stopped and terminate/2 was called before that change
feed died, then

```
foreach(fun({_Tag, Pid}) -> ... end, [StartPids])
```

would crash with a function clause error.

Make sure add the replicator db name to the changes feed.

Jira: COUCHDB-3082
